File: app/schemas/collection.py
"""Schémas Pydantic pour le module ProductCollection."""
from typing import Optional
from app.schemas.base import BaseSchema
from app.schemas.product import ProductList


class CollectionBase(BaseSchema):
    name: str
    description: Optional[str] = None
    is_active: bool = True


class CollectionCreate(CollectionBase):
    pass


class CollectionUpdate(BaseSchema):
    name: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None


class CollectionResponse(CollectionBase):
    id: int
    tenant_id: int


class CollectionWithProducts(CollectionResponse):
    products: list[ProductList] = []


class CollectionAddProducts(BaseSchema):
    product_ids: list[int]


class CollectionList(BaseSchema):
    items: list[CollectionResponse]
    total: int

File: frontend/apps/marveline/src/types/collection.ts
export interface Collection {
  id: number
  tenant_id: number
  name: string
  description: string | null
  is_active: boolean
}

export interface CollectionWithProducts extends Collection {
  products: { id: number; name: string; reference?: string }[]
}

export interface CollectionCreate {
  name: string
  description?: string
  is_active?: boolean
}

export interface CollectionUpdate {
  name?: string
  description?: string
  is_active?: boolean
}

export interface CollectionListResponse {
  items: Collection[]
  total: number
}
